pkg/httpsproxy/proxylet/request_handler_impl.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) The One True Way 2020. Apache License 2.0. The authors accept no liability, 0 nada for the use of this software.  It is offered "As IS"  Have fun with it!!
+ */
+package proxylet
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"net/http"
+
+	"github.com/nats-io/nats.go"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/theotw/natssync/pkg/httpsproxy/models"
+	"github.com/theotw/natssync/pkg/httpsproxy/server"
+)
+
+type requestHandler struct {
+	counter    int
+	httpClient *http.Client
+	natsClient *nats.Conn
+	locationID string
+}
+
+func NewRequestHandler(LocationID string, natsClient *nats.Conn) *requestHandler {
+
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	return &requestHandler{
+		counter:    0,
+		httpClient: &http.Client{Transport: transport},
+		natsClient: natsClient,
+		locationID: LocationID,
+	}
+}
+
+func (rh *requestHandler) SetLocationID(locationID string) {
+	rh.locationID = locationID
+}
+
+func (rh *requestHandler) HttpHandler(m *nats.Msg) {
+	if string(m.Data) == "" {
+		return
+	}
+
+	log.Printf("[#%d] Received on [%s]: '%s'", rh.counter, m.Subject, string(m.Data))
+
+	var resp *server.HttpApiResponseMessage
+
+	req, err := server.NewHttpApiReqMessageFromNatsMessage(m)
+	if err != nil {
+		log.WithError(err).Errorf("Error decoding http message")
+		resp = server.NewHttpApiResponseMessageFromError(err)
+		_ = rh.natsClient.Publish(m.Reply, []byte("ack"))
+		_ = rh.natsClient.Flush()
+		return
+	}
+
+	httpResp, err := rh.httpClient.Do(req.ToHttpRequest())
+	if err != nil {
+		log.WithError(err).Errorf("Error decoding http message")
+		resp = server.NewHttpApiResponseMessageFromError(err)
+	} else {
+		resp = server.NewHttpApiResponseMessageFromHttpResponse(httpResp)
+	}
+
+	respBytes, err := json.Marshal(&resp)
+	if err != nil {
+		log.WithError(err).Error("Error marshaling response body")
+	}
```

Comment:
Shouldn't we `return` after logging the error? Otherwise, we'll attempt to publish the failed marshal down below.

Reply:
yup, I tried to make sure that this was translated correctly. I guess I don't fully understand the protocol, 
why do we send out an `ack` if we fail to decode the request ? 
https://github.com/theotw/natssync/pull/47/files/b419996607c37421e67586b0c184b4416793a891#diff-f3ae9a04639e373f0508976d63c4334e2c87770bf1bb2fe658edf9e602fc9668R54-R60

vs error response here ? 
https://github.com/theotw/natssync/pull/47/files/b419996607c37421e67586b0c184b4416793a891#diff-f3ae9a04639e373f0508976d63c4334e2c87770bf1bb2fe658edf9e602fc9668R66

if we fail to Marshal the response, we will end up with: 
`rh.natsClient.Publish(m.Reply, "")`
should I be replacing it with : `resp = server.NewHttpApiResponseMessageFromError(err)` ?

Reply:
That sounds good to me. @bmason42 do you have any input here?